common/tier4_planning_rviz_plugin/include/tier4_planning_rviz_plugin/path/display_base.hpp

```diff
@@ -232,6 +201,43 @@ class AutowarePathBaseDisplay : public rviz_common::MessageFilterDisplay<T>
   }
 
 protected:
+  std::unique_ptr<Ogre::ColourValue> gradation(
+    const QColor & color_min, const QColor & color_max, const double ratio)
+  {
+    std::unique_ptr<Ogre::ColourValue> color_ptr(new Ogre::ColourValue);
+    color_ptr->g =
+      static_cast<float>(color_max.greenF() * ratio + color_min.greenF() * (1.0 - ratio));
+    color_ptr->r = static_cast<float>(color_max.redF() * ratio + color_min.redF() * (1.0 - ratio));
+    color_ptr->b =
+      static_cast<float>(color_max.blueF() * ratio + color_min.blueF() * (1.0 - ratio));
+
+    return color_ptr;
+  }
+
+  std::unique_ptr<Ogre::ColourValue> setColorDependsOnVelocity(
+    const double vel_max, const double cmd_vel)
+  {
+    const double cmd_vel_abs = std::fabs(cmd_vel);
+    const double vel_min = 0.0;
+
+    std::unique_ptr<Ogre::ColourValue> color_ptr(new Ogre::ColourValue());
+    if (vel_min < cmd_vel_abs && cmd_vel_abs <= (vel_max / 2.0)) {
+      double ratio = (cmd_vel_abs) / (vel_max / 2.0);
+      color_ptr = gradation(property_min_color_.getColor(), property_mid_color_.getColor(), ratio);
+    } else if ((vel_max / 2.0) < cmd_vel_abs && cmd_vel_abs <= vel_max) {
+      double ratio = (cmd_vel_abs - vel_max / 2.0) / (vel_max / 2.0);
+
+      color_ptr = gradation(property_mid_color_.getColor(), property_max_color_.getColor(), ratio);
+    } else if (vel_max < cmd_vel_abs) {
+      // Use max color when velocity exceeds max
+      *color_ptr = rviz_common::properties::qtToOgre(property_max_color_.getColor());
+    } else {
+      // Use min color when velocity is below min
+      *color_ptr = rviz_common::properties::qtToOgre(property_min_color_.getColor());
+    }
+
+    return color_ptr;
+  }
```

Comment:
Could you turn:
- `color_min`
- `color_mid`
- `color_max`
into function parameters instead of directly using them here?

Then you can move it back up instead of making it a protected member of the class.

Reply:
Applied with: https://github.com/autowarefoundation/autoware.universe/pull/8972/commits/d8ba16f9a28d27cc156a64f8cd64e15b0e96d58c